hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -236,89 +247,157 @@ object HoodieSparkSqlWriter {
-            val shouldCombine = parameters(INSERT_DROP_DUPS.key()).toBoolean ||
-              operation.equals(WriteOperationType.UPSERT) ||
-              parameters.getOrElse(HoodieWriteConfig.COMBINE_BEFORE_INSERT.key(),
-                HoodieWriteConfig.COMBINE_BEFORE_INSERT.defaultValue()).toBoolean
-            val hoodieAllIncomingRecords = genericRecords.map(gr => {
-              val processedRecord = getProcessedRecord(partitionColumns, gr, dropPartitionColumns)
-              val hoodieRecord = if (shouldCombine) {
-                val orderingVal = HoodieAvroUtils.getNestedFieldVal(gr, hoodieConfig.getString(PRECOMBINE_FIELD), false, parameters.getOrElse(
-                  DataSourceWriteOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.key(),
-                  DataSourceWriteOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.defaultValue()).toBoolean)
-                  .asInstanceOf[Comparable[_]]
-                DataSourceUtils.createHoodieRecord(processedRecord,
-                  orderingVal,
-                  keyGenerator.getKey(gr),
-                  hoodieConfig.getString(PAYLOAD_CLASS_NAME))
-              } else {
-                DataSourceUtils.createHoodieRecord(processedRecord, keyGenerator.getKey(gr), hoodieConfig.getString(PAYLOAD_CLASS_NAME))
+              sparkContext.getConf.registerAvroSchemas(schema)
+              log.info(s"Registered avro schema : ${schema.toString(true)}")
+
+              // Convert to RDD[HoodieRecord]
+              val genericRecords: RDD[GenericRecord] = HoodieSparkUtils.createRdd(df, structName, nameSpace, reconcileSchema,
+                org.apache.hudi.common.util.Option.of(schema))
+              val shouldCombine = parameters(INSERT_DROP_DUPS.key()).toBoolean ||
+                operation.equals(WriteOperationType.UPSERT) ||
+                parameters.getOrElse(HoodieWriteConfig.COMBINE_BEFORE_INSERT.key(),
+                  HoodieWriteConfig.COMBINE_BEFORE_INSERT.defaultValue()).toBoolean
+              val writeSchema =
+                if (dropPartitionColumns) generateSchemaWithoutPartitionColumns(partitionColumns, schema)
+                else schema
+
+              // init write client
+              val client = hoodieWriteClient.getOrElse(
+                DataSourceUtils.createHoodieClient(jsc, writeSchema.toString, path, tblName, mapAsJavaMap(parameters))
+              ).asInstanceOf[SparkRDDWriteClient[HoodieRecordPayload[Nothing]]]
+              // needed to generate commit sequence number
+              val partitionId = client.getEngineContext.getTaskContextSupplier.getPartitionIdSupplier.get
+              val recordIndex = new AtomicLong(1)
+
+              val hoodieAllIncomingRecords = genericRecords.map(gr => {
+                val processedRecord = getProcessedRecord(partitionColumns, gr, dropPartitionColumns)
+                val csn = HoodieRecord.generateSequenceId(instantTime, partitionId, recordIndex.getAndIncrement())
```

Comment:
I thought plan was to re-use the commit seq no that we generate within the writer (i.e. by executors). or is my understanding wrong.